### PR TITLE
building the radial symmetry functions

### PR DESCRIPTION
**Pull request by jeherr**
Code style fixes

Just a few cosmetic fixes here and there to comply with python standards

Fix to MSet loading

I believe this fixes a bug in the loading where each Geometry may have had a bug in loading the properties dictionary correctly

build radial two-body pieces

builds the radial features and scatters them into element channels

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -101,6 +101,8 @@ ENV/
 # mypy
 .mypy_cache/
 
-# h5py files
+# data files used for TensorChem
 *.h5
 *.json
+*.mset
+*.dset
```
Comment (alexisdriscoll) [inline .gitignore:108]: I don't think we need .dset, I named one of the files this but I ended up deleting it

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # How to contribute
 
 We welcome contributions from external contributors, and this document
-describes how to merge code changes into this tensormol_jax. 
+describes how to merge code changes into this tensorchem.
```
Comment (alexisdriscoll) [inline .github/CONTRIBUTING.md:4]: Does everything automatically update if the repo is renamed?

File: tensorchem/dataset/dataset.py [hunk 3]
```diff
@@ -4,11 +4,14 @@
 """
 
 import json
-import numpy as np 
+from typing import Tuple
```
Comment (alexisdriscoll) [inline tensorchem/dataset/dataset.py:7]: Where do you use this "Tuple" function?

File: tensorchem/dataset/dataset.py [hunk 4]
```diff
@@ -131,13 +134,27 @@ def from_mset(cls, msets):
         mixed_data = cls()
         for mset in msets:
             for geom in mset.geometries:
-                sample = {"atomic_number": mset.atomic_nums, "coordinates": geom.coords.tolist()}
+                sample = {"atomic_numbers": mset.atomic_nums, "coordinates": geom.coords}
```
Comment (alexisdriscoll) [inline tensorchem/dataset/dataset.py:137]: I thought the "coords" property had to be named saved with .tolist() because it is stored as a numpy array in the dataset.

File: tensorchem/dataset/dataset.py [hunk 5]
```diff
@@ -131,13 +134,27 @@ def from_mset(cls, msets):
...
                 sample.update({key: value for key, value in geom.properties.items()})
-                sample.update({key: value for key, value in mset.identifiers.items()})
                 mixed_data.samples.append(sample)
         return mixed_data
 
 
+class Sample:
```
Comment (alexisdriscoll) [inline tensorchem/dataset/dataset.py:143]: Where/why are you using this?

File: tensorchem/dataset/molecule.py [hunk 6]
```diff
@@ -36,30 +36,49 @@ def save(self, filename=None):
...
+        self.identifiers = json_data['identifiers']
+        for i, coords in enumerate(json_data['coordinates']):
+            properties = {key: value[i] for key, value in json_data['properties'].items()}
+            self.geometries.append(Geometry(self.atomic_nums, coords, properties))
 
 
 class Geometry:
-    def __init__(self, coords=None, properties=None):
+    def __init__(self, atomic_nums, coords, properties=None):
+        self.atomic_nums = ()
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:64]: Why are you adding this?

File: tensorchem/dataset/molecule.py [hunk 7]
```diff
@@ -36,30 +36,49 @@ def save(self, filename=None):
...
         with open(filename) as f:
             json_data = json.load(f)
         for key in json_data.keys():
             if type(json_data[key]) == dict:
                 self.identifiers = {key: json_data[key]}
         self.atomic_nums = json_data['atomic_number']
-        self.geometries = [Geometry(np.array(coords), {key: value for key, value in json_data['properties'].items()}) for coords in json_data['coordinates']]
+        self.identifiers = json_data['identifiers']
+        for i, coords in enumerate(json_data['coordinates']):
+            properties = {key: value[i] for key, value in json_data['properties'].items()}
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:58]: Thanks for fixing this.

File: tensorchem/dataset/molecule.py [hunk 8]
```diff
@@ -36,30 +36,49 @@ def save(self, filename=None):
...
+    def __repr__(self):
+        rep = str(self.n_atoms)
+        rep += "\n\n"
+        for i, at_num in enumerate(self.atomic_nums):
+            rep += "     ".join((str(at_num), str(self.coords[i, 0]), str(self.coords[i, 1]), str(self.coords[i, 2])))
+            rep += "\n"
+        return rep
+
+    @property
+    def n_atoms(self):
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:80]: Should we do this for the other properties too?

File: tensorchem/dataset/molecule.py [hunk 9]
```diff
@@ -36,30 +36,49 @@ def save(self, filename=None):
...
+    def __init__(self, atomic_nums, coords, properties=None):
+        self.atomic_nums = ()
         self.coords = coords
-        self.properties = properties
+        if properties is None:
+            self.properties = {}
+        else:
+            self.properties = properties
+
+    def __repr__(self):
```
Comment (alexisdriscoll) [inline tensorchem/dataset/molecule.py:71]: I don't understand this at all haha

File: tensorchem/util/ani_dataloader.py [hunk 10]
```diff
@@ -79,18 +79,16 @@ def load_ani1x(path_to_h5file, data_keys=[]):
...
-        #exit(0)
-        msets.append(mset)
-    return msets
+        mset.save()
+    return
 
 
 if __name__ == "__main__":
-    path_to_h5file = '/home/adriscoll/tensormol-jax/tensormol_jax/data/ani1x-release.h5'
+    path_to_h5file = '/Users/johnherr/tensorchem/tensorchem/data/ani1x-release.h5'
```
Comment (alexisdriscoll) [inline tensorchem/util/ani_dataloader.py:91]: this looks much better

File: tensorchem/tests/test_dataset.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import sys
+import numpy as np
+
+from tensorchem.dataset.dataset import MixedDataset
+from tensorchem.dataset.molecule import MoleculeSet
+
+
+def test_len_MixedDataset():
+    mixed_data = MixedDataset()
+    mixed_data.load('tensorchem/tests/h2o.dset')
```
Comment (alexisdriscoll) [inline tensorchem/tests/test_dataset.py:12]: Why are you using .dset files?

File: tensorchem/featurizers/symmetry_functions.py [hunk 12]
```diff
@@ -0,0 +1,34 @@
+"""
```
Comment (alexisdriscoll) [inline tensorchem/featurizers/symmetry_functions.py:1]: Next 3 files look good to me

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (alexisdriscoll) [reply .github/CONTRIBUTING.md:4]: A lot of the following changes are just renaming.

File: .gitignore (hunk 1 above)
Comment (jeherr) [reply .gitignore:108]: I changed the dataset objects to save to their own filetype. It should be different from the MoleculeSet objects, because it may not contain all of the data from the MoleculeSet, and the data should be stored in a slightly different format.

The way I envision the interplay between MoleculeSet and Dataset is that MoleculeSet will work more like repositories of conformations/different geometries of the same molecule, and a dictionary of labels like energies, forces, etc. but each sample may have energies from multiple functional/basis set calculations.

When we build a dataset from MoleculeSets, then we feed the coordinates, atomic numbers, and any labels we want to use for that particular dataset. This may not be all of the labels, because we may only be training with certain ones for a particular example. The dataset can then save itself to a distinct file from the MSet because it only contains part of the data, and can be offloaded more easily to a computer server for training.

So for this reason I think we should make their filetypes unique. Just to tell them apart.

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (jeherr) [reply .github/CONTRIBUTING.md:4]: Lol no I went through and changed this all by just doing a search and replace. I got tired of typing tensormol_jax so I decided to go ahead and get it changed. I hope it doesn't mess up any of the testing, codecov, docs, etc from the CookieCutter.

File: tensorchem/dataset/dataset.py (hunk 3 above)
Comment (jeherr) [reply tensorchem/dataset/dataset.py:7]: Good catch. I think I put this there because I started to use it but changed my mind. Will remove it.

File: tensorchem/dataset/dataset.py (hunk 4 above)
Comment (jeherr) [reply tensorchem/dataset/dataset.py:137]: So I was loading an MSet I saved with the ani_dataloader to use this class method because I wasn't passing one of the tests you wrote. What was happening was the coords variable was already a list, so it had no tolist() method.

We should make sure that we have well defined types for the variables. So in Geometry right now we don't specify whether coords should be a list, or a numpy array, or anything really.

We're also not saving the dataset with this routine. This method is only creating a dataset object. It could be that we will want to use that dataset object immediately, in which case coords should be in the torch tensor format.

File: tensorchem/dataset/dataset.py (hunk 5 above)
Comment (jeherr) [reply tensorchem/dataset/dataset.py:143]: Haha yeah I haven't used this yet at all. As I was trying to update some things I was running into some issues with handling the data. I was considering adding a "Sample" class for the Dataset classes to act like the Geometry class does for MoleculeSet objects. A Sample would be one training/testing sampling that belongs to the Dataset.

I'm not 100% I will end up using this or not, so don't worry about it for now. I really shouldn't be adding this to master unless it's complete too.

File: tensorchem/dataset/molecule.py (hunk 8 above)
Comment (jeherr) [reply tensorchem/dataset/molecule.py:80]: Only when it makes sense. The reason I did this here is because if we set n_atoms like other properties, it would be set to a constant value. Imagine we take a geometry object which already has a set value for atomic_numbers and n_atoms properties. Now if we change atomic_numbers, the number of atoms could change, and we would have to go update n_atoms as well. This way, every time we call Geometry.n_atoms, it automatically checks the len(atomic_numbers) to give us the most up to date number of atoms for that Geometry. We no longer have to remember to update n_atoms when we update atomic_numbers.

File: tensorchem/dataset/molecule.py (hunk 6 above)
Comment (jeherr) [reply tensorchem/dataset/molecule.py:64]: It has to do with the n_atoms property decorator I added to this class. I know it's not how we discussed it before, but I needed a way to get the number of atoms for each geometry. I think there was another reason too. I'm not happy with how I have implemented this part now so it's definitely subject to change. I just don't know what is best right now.

File: tensorchem/dataset/molecule.py (hunk 9 above)
Comment (jeherr) [reply tensorchem/dataset/molecule.py:71]: For your tests, one of them was that the type of Geometry.properties be a dictionary. If a Geometry is created without giving it a properties dictionary, then properties was NoneType instead of the DictType.

File: tensorchem/util/ani_dataloader.py (hunk 10 above)
Comment (jeherr) [reply tensorchem/util/ani_dataloader.py:91]: Lol yes much shorter!
